src/strict_fstream.hpp

```diff
@@ -51,17 +50,20 @@ static std::string strerror()
 {
     std::string buff(256, '\0');
 #ifdef _WIN32
-    if (strerror_s(buff.data(), buff.size(), errno) != 0) {
+    // Since strerror_s might set errno itself, we need to store it.
+    const int err_num = errno;
+    if (strerror_s(buff.data(), buff.size(), err_num) != 0) {
         return trim_to_null(buff);
     } else {
-        return "Unknown error";
+        return "Unknown error (" + std::to_string(err_num) + ")";
     }
-#elif ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600 || defined(__APPLE__)) && ! _GNU_SOURCE) || defined(BROKEN_GNU_STRERROR_R)
+#elif ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600 || defined(__APPLE__)) && ! _GNU_SOURCE) || defined(__MUSL__)
 // XSI-compliant strerror_r()
-    if (strerror_r(errno, buff.data(), buff.size()) == 0) {
+    const int err_num = errno; // See above
+    if (strerror_r(err_num, buff.data(), buff.size()) == 0) {
```

Comment:
This doesn't compile on MacOS / Alpine due to `const char*` to `char*` caused by `buff.data()`

Reply:
I'm confused, does std::string::data() on macos return a const for some reason?

Reply:
doh, forgot that that was fixed in c++17

Reply:
and the reason I don't have macos CI set up is because apple has so outdated C++ support, unfortunately...